src/Uuid.php

```diff
@@ -514,6 +514,22 @@ public static function fromDateTime(
         return self::getFactory()->fromDateTime($dateTime, $node, $clockSeq);
     }
 
+    /**
+     * Creates a UUID from the Hexadecimal object
+     *
+     * @param Hexadecimal $hex Hexadecimal object representing a hexadecimal number
+     *
+     * @return UuidInterface A UuidInterface instance created from the Hexadecimal
+     * object representing a hexadecimal number
+     *
+     * @psalm-pure note: changing the internal factory is an edge case not covered by purity invariants,
+     *             but under constant factory setups, this method operates in functionally pure manners
+     */
+    public static function fromHexadecimal(Hexadecimal $hex): UuidInterface
+    {
+        return self::getFactory()->fromHexadecimal($hex);
```

Comment:
This is failing static analysis because `getFactory()` returns a `UuidFactoryInterface`, which doesn't have this method on it. This can cause problems if using a factory that doesn't have this method.

```suggestion
        $factory = self::getFactory();
        
        if (\method_exists($factory, 'fromHexadecimal')) {
            return self::getFactory()->fromHexadecimal($hex);
        }

        throw new \BadMethodCallException('The method fromHexadecimal() does not exist on the provided factory');
```

Reply:
I'm going to commit this suggestion, though it might still fail coding standards checks because of the use of the root namespace without `use` statements. I'll fix those, though.